execute/row_fn.go

```diff
@@ -22,15 +24,20 @@ type dynamicFn struct {
 
 	recordCols map[string]int
 	references []string
+
+	deps dependencies.Interface
+	ctx  context.Context
 }
 
-func newDynamicFn(fn *semantic.FunctionExpression, scope compiler.Scope) dynamicFn {
+func newDynamicFn(fn *semantic.FunctionExpression, scope compiler.Scope, ctx context.Context, deps dependencies.Interface) dynamicFn {
 	return dynamicFn{
 		compilationCache: compiler.NewCompilationCache(fn, scope),
 		inRecord:         values.NewObject(),
 		recordName:       fn.Block.Parameters.List[0].Key.Name,
 		references:       findColReferences(fn),
 		recordCols:       make(map[string]int),
+		deps:             deps,
+		ctx:              ctx,
```

Comment:
Similar question here, should we move the context and deps off the struct but rather pass them in via `Eval` each time?

In general its best practice to not store a context on the heap (i.e. a struct). This is because by definition context carries only information about the current call stack so storing it on the heap instead of the stack creates opportunity for bugs. For example  if context lives on the heap beyond the stack lifetime how do you ensure it gets cleaned up? If you always keep it on the stack (i.e. a function parameter) you avoid this whole class of bugs.